terraform/main.tf

```diff
@@ -1,11 +1,20 @@
 locals {
-  hws = toset([
-    "hw0",
-    "hw1",
-    "hw2",
-    "hw3",
-    "hw4",
-    "final_project",
+  // Name of the homework mapped to published status
+  published = {
+    hw0           = false
+    hw1           = false
+    hw2           = false
+    hw3           = false
+    hw4           = false
+    final_project = false
+  }
+  // Create a list of objects representing all homework repos to be (eventually) created. 
+  hws = flatten([
+    for hw, published in local.published : [
+      for student in var.students : {
+        hw = hw, student = student, published = published
```

Comment:
this shit is SO cursed

Reply:
My favorite part about this is that it's officially recommended